[29기 박무선] 메인 페이지 레이아웃 & 이미지 슬라이더 기본 레이아웃

mooseonpark opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [ ] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 메인 페이지 레이아웃 작업
- 캐러셀(이미지 슬라이더) 레이아웃 작업

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 메인 페이지 레이아웃
- 레이아웃 구성

2. 캐러셀 레이아웃
- 반복되는 캐러셀 컴포넌트화 및 map함수 사용하여 중복되는 코드 재사용
- styled component 작업 중


<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- git rebase 사용법에 익숙해지려고 노력중입니다.
- styled component 사용이 기존 sass 작업과 헷갈리지만 참고하며 작업중입니다.
- 처음으로 라이브러리를 사용하여 작업의 효율성을 높일 수 있었습니다. 

<br />

## :: 기타 질문 및 특이 사항
-

yeonuk-hwang commented on src/pages/Main/JobInfo/JobInfo.js:
```diff
@@ -0,0 +1,35 @@
+function JobInfo() {
+  return (
+    <Parent>
+      {JOB_INFOS.map(JOB => (
```
```suggestion
      {JOB_INFOS.map(job => (
```

인자로 들어오는 값은 상수가 아니니까 일반적인 네이밍 컨벤션으로 적용하시는게 좋아보이네요!

yeonuk-hwang commented on public/index.html:
```diff
@@ -12,7 +12,20 @@
     />
   </head>
 
+<<<<<<< HEAD
```
conflict 꼼꼼하게 확인해주세요!

yeonuk-hwang commented on src/pages/Main/ImageSlider/FirstSlider.js:
```diff
@@ -0,0 +1,53 @@
+
+export default FirstSlider;
+
+const Container = styled.div`
+  margin: 0 auto;
+  width: 500px;
+`;
+const StyleSlide = styled(Slider)`
+  .slick-list {
+    justify-content: center;
+    height: 300px;
+    background-color: darkgoldenrod;
+  }
+`;
```
```suggestion
const Container = styled.div`
  margin: 0 auto;
  width: 500px;
`;

const StyleSlide = styled(Slider)`
  .slick-list {
    justify-content: center;
    height: 300px;
    background-color: darkgoldenrod;
  }
`;
```
css에서 다른 selector들 끼리는 공백을 부여해주세요!
사소한 부분이라고 간과할 수 있지만 이런 부분들이 모여서 코드의 가독성을 좋게 만듭니다

yeonuk-hwang commented on src/pages/Main/ImageSlider/FirstSlider.js:
```diff
@@ -0,0 +1,53 @@
+  return (
+    <Container>
+      <StyleSlide {...settings}>
+        {FIRST_DATAS.map(DATA => (
```
First Data 같은 이름보다는 명확히 의미가 드러나는 이름으로 작명해주세요!

yeonuk-hwang reviewed (changes requested):
고생하셨습니다 무선님 리뷰 확인 및 반영해주세요~!

yeonuk-hwang commented:
PR 본문에 제일 상단에 커밋메세지들이 그대로 남아있는 것 같네요 수정해주세요!

yeonuk-hwang commented on src/pages/Main/BannerItem/BannerItem.js:
```diff
@@ -0,0 +1,37 @@
+function BannerItem() {
+  return (
+    <Item>
+      {BANNERS_DATA.map(BANNER => (
```
```suggestion
      {BANNERS.map(banner => (
```

인자로 들어오는 값은 계속 바뀌기 때문에 상수로 사용하지 않습니다. 그리고 _DATA 등의 suffix는 굳이 안붙여도 되니까 안붙이는것도 좋습니다!

그리고 매개변수자리에서 banner 바로 구조분해할당하는것도 괜찮아보이네요!

mooseonpark replied:
배너 부분은 바뀌는 데이터가 아니라서 상수데이터로 두었는데 수정해야할까요?

yeonuk-hwang replied:
callback의 인자는 변하는 값이니까 상수로 네이밍하는건 좋지 않을 것 같네요!

yeonuk-hwang reviewed (changes requested):
무선님 리뷰 반영 및 라벨 수정헤주세요, 라운지에서 피드백 드린 부분이랑, 코멘트상에서 피드백 드린 부분 전반적으로 다 적용될 수 있을 것 같아서 전반적으로 적용한번 해주세요!

mooseonpark committed:
Modify: 리뷰 반영
Modify: main 수정중
Modify: main 수정
Fix: conflict resolved
Fix: conflict resolved
Modify: 리뷰 반영
Modify: main 수정중
Modify: main 수정
Modify: conflict resolved
Modify: 리뷰 반영
Fix: .env 파일 삭제
Modify: 수정
Modify: styled component 수정
Fix: env 파일 삭제

yeonuk-hwang commented on .env:
```diff
@@ -0,0 +1 @@
+REACT_APP_KAKAO_APP_KEY=746a759a33686fbf7cf478afc0274961
```
gitignore에 등록해주세요! github상에 올라오면 안됩니다!

yeonuk-hwang reviewed (changes requested):
무선님 .env만 처리해주세요!

yeonuk-hwang reviewed (changes requested):
무선님 env 파일 계속 올라와있네요!

- .gitignore에 등록하기 전에 한번이라도 git으로 관리해버린 파일들은(add, commit 등) 계속해서 git에 의해서 추적이 됩니다.
- 이를 없애기 위해서는 git의 기존 캐쉬를 삭제해줘야합니다.
- 아래의 명령어를 통해 캐쉬 삭제해주시고 git으로 추적 안되는 것 확인해주세요
- `git rm --cached 파일이름`

코멘트 참고하셔서 수정해주시고 안올라왔는지 파일체인지 확인해주세요!

yeonuk-hwang commented:
rebase도 진행해주세요!

yeonuk-hwang commented:
commit 하나로 합쳐주세요!!
